presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py

```diff
@@ -634,7 +634,49 @@ def _get_text_metadata(
         return metadata_text, is_name, is_patient
 
     @staticmethod
-    def _process_names(text_metadata: list, is_name: list) -> list:
+    def augment_word(word: str) -> list:
+        """Apply multiple types of casing to the provided string.
+
+        :param words: String containing the word or term of interest.
+
+        :return: List of the same string with different casings and spacing.
+        """
+        word_list = []
+        if word != "":
+            word_list = []
+
+            # Replacing separator character with space, if any
+            text_1 = word.replace("^", " ")
+            text_1 = text_1.replace("-", " ")
+
+            # Capitalize all characters in string
+            text_2 = text_1.upper()
+
+            # Lowercase all characters in string
+            text_3 = text_1.lower()
+
+            # Capitalize first letter in each part of string
+            text_4 = text_1.title()
+
```

Comment:
would it make sense to match this from the other side?
The client of word_list would match lower() to lower()?

Reply:
I included `.lower()` and the other casing transformations inside this method such that it could be used by (1) the DICOM image redactor engine when metadata is used and (2) users who wish to call the method directly to augment any list of words they may have (e.g., allow list).

Reply:
In that case, it might even be better to not reduce it to lowecase.
So that in (2), users that care about casing don't lose that information and can decide if they want to exact match or do .lower()
the DICOM image redactor engine can do .lower() itself on the input vs each element in the returned list.  
Or am I missing something?
Anyway, up to you

Reply:
Good point. I have added the bool argument `case_sensitive` such that users can specify as needed.